v19.2/movr.md

```diff
@@ -0,0 +1,85 @@
+- The `movr` dataset, which consists of rows of data that populate tables in the `movr` database. The `movr` dataset is built into `cockroach demo` and `cockroach workload`.
+- The MovR application is a fully-functional vehicle-sharing application, written in Python. All of MovR application source code is open-source, and available on the [movr](https://github.com/cockroachdb/movr) GitHub repository.
+
+## The `movr` Database
+
+Three tables in the `movr` database store most of the user, vehicle, and ride data for MovR:
+
+- `users` represents the people registered for the service.
+- `vehicles` represents the pool of vehicles for the service.
+- `rides` represents when and where users have rented a vehicle.   
+
+Each table has a composite primary key of `city` and `id`, with `city` being first in the key.
+
+<img src="{{ 'images/v19.2/geo-partitioning-schema.png' | relative_url }}" alt="Geo-partitioning schema" style="max-width:100%" />
+
+The `movr` database also contains the `promo_codes`, `user_promo_codes`, and `vehicle_location_histories` tables.
+
+## Using `cockroach` Commands to Generate Schemas and Data for MovR
+
+Use `cockroach` commands to load the `movr` database and dataset that are built into the CockroachDB binary.
+
+[`cockroach demo movr`](cockroach-demo.html) opens a temporary, in-memory cluster with the `movr` database preloaded and set as the [current database](sql-name-resolution.html#current-database):
+
+[`cockroach workload init movr`](cockroach-workload.html) loads the `movr` database and some sample data to a running cluster.
+
+## Using the MovR Application
+
+The workflow for MovR is as follows:
+
+1. A user loads the app and sees the 25 closest vehicles. Behind the scenes, this is a `SELECT` from the `vehicles` table:
+
+    ~~~ sql
+    > SELECT id, city, status, ... FROM vehicles WHERE city = <user location>
+    ~~~
+
+2. The user signs up for the service, which is an `INSERT` of a row into the `users` table:
+
+    ~~~ sql
+    > INSERT INTO users (id, name, address, ...) VALUES ...
+    ~~~
+
+3. In some cases, the user adds their own vehicle to share, which is an `INSERT` of a row into the `vehicles` table:
+
+    ~~~ sql
+    > INSERT INTO vehicles (id, city, type, ...) VALUES ...
+    ~~~
+
+4. More often, the user reserves a vehicle and starts a ride, which is an `UPDATE` of a row in the `vehicles` table and an `INSERT` of a row into the `rides` table:
+
+    ~~~ sql
+    > UPDATE vehicles SET status = 'in_use' WHERE ...
+    ~~~
+
+    ~~~ sql
+    > INSERT INTO rides (id, city, start_addr, ...) VALUES ...
+    ~~~
+
+5. The user ends the ride and releases the vehicle, which is an `UPDATE` of a row in the `vehicles` table and an `UPDATE` of a row in the `rides` table:
+
+    ~~~ sql
+    > UPDATE vehicles SET status = 'available' WHERE ...
+    ~~~
+
+    ~~~ sql
+    > UPDATE rides SET end_address = <value> ...
+    ~~~
```

Comment:
This section is great!  The mapping of use cases to SQL is so easy to follow / understand.

Reply:
This was shamelessly copied from https://www.cockroachlabs.com/docs/stable/demo-geo-partitioning.html ! :)